lib/vm/src/instance.rs

```diff
@@ -963,11 +1094,15 @@ impl InstanceHandle {
     /// It should ideally return `NonNull<Instance>` rather than
     /// `NonNull<u8>`, however `Instance` is private, and we want to
     /// keep it private.
-    pub fn allocate_instance(module: &ModuleInfo) -> (NonNull<u8>, VMOffsets) {
+    pub fn allocate_instance(
+        module: &ModuleInfo,
+    ) -> (
+        UnpreparedInstanceAllocatorSetter,
+        Vec<NonNull<VMMemoryDefinition>>,
+        Vec<NonNull<VMTableDefinition>>,
+    ) {
         let offsets = VMOffsets::new(mem::size_of::<*const u8>() as u8, module);
-        let (instance_ptr, _instance_layout) = InstanceAllocator::allocate_instance(&offsets);
-
-        (instance_ptr.cast(), offsets)
+        UnpreparedInstanceAllocatorSetter::new(offsets)
     }
```

Comment:
Can we move this function to the `UnpreparedInstance` struct?

Reply:
I thought it'd be more discoverable there but I think it's fine to put it on the type itself too